src/components/Pill/PillButton/__tests__/PillButton.spec.tsx

```diff
@@ -9,51 +11,71 @@ describe('<PillButton> component', () => {
         beforeEach(() => {
-            wrapper = mount(
+            view = render(
                 <PillButton toggleDropdown={toggleDropdownMock} onClear={onClearMock} name={name} />
             );
         });
 
         it('should render correctly', () => {
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('.PillButton__pill--isActive')).toHaveLength(0);
-            expect(wrapper.find('.PillButton__pill--isOpen')).toHaveLength(0);
+            expect(view.container).toMatchSnapshot();
+            expect(screen.getByRole('button', { name: '' })).not.toHaveClass(
+                'PillButton__button PillButton__button--isOpen'
+            );
+            expect(screen.getByRole('button', { name: `${name}` })).not.toHaveClass(
+                'PillButton__pill PillButton__pill--isOpen PillButton__pill--isActive'
+            );
         });
-        it('should trigger toggle state once when clicked', () => {
-            wrapper.find('.PillButton__pill').simulate('click');
+
+        it('should trigger toggle state once when clicked', async () => {
+            const user = userEvent.setup();
+            await user.click(screen.getByRole('button', { name: `${name}` }));
+
             expect(toggleDropdownMock).toHaveBeenCalledTimes(1);
         });
-        it('should trigger toggle state once on keyboard interaction', () => {
-            wrapper
-                .find('.PillButton__pill')
-                .simulate('focus')
-                .simulate('keyDown', { key: ENTER_KEY });
+
+        it('should trigger toggle state once on keyboard interaction', async () => {
+            const user = userEvent.setup();
+            const button = screen.getByRole('button', { name: `${name}` });
+            button.focus();
+            await user.keyboard(`[${ENTER_KEY}]`);
+
             expect(toggleDropdownMock).toHaveBeenCalledTimes(1);
         });
+
         it('should have arrow down label', () => {
-            expect(wrapper.find('IoIosArrowDown')).toHaveLength(1);
+            expect(screen.getByRole('img')).toBeInTheDocument();
+            expect(screen.getByRole('img')).toHaveClass('PillButton__arrowIcon');
         });
-        it('should trigger toggle state once when button is clicked', () => {
-            wrapper.find('button').simulate('click');
+
+        it('should trigger toggle state once when button is clicked', async () => {
+            const user = userEvent.setup();
+            const button = screen.getByRole('button', { name: `${name}` });
+            await user.click(button);
+
             expect(toggleDropdownMock).toHaveBeenCalledTimes(1);
         });
-        it('should trigger toggle state once on keyboard interaction with button', () => {
-            wrapper.find('button').simulate('focus').simulate('keyDown', { key: ENTER_KEY });
+
+        it('should trigger toggle state once on keyboard interaction with button', async () => {
```

Comment:
it looks like a repetition of the test from line 44.

Reply:
yes :+1: